### ISSUE TITLE
out/soong/.intermediates/frameworks/av/drm/libmediadrm/libmediadrm/android_arm_armv7-a-neon_cortex-a15_shared_core/gen/proto/frameworks/av/drm/libmediadrm/protos/plugin_metrics.pb.cc

### ISSUE THREAD

**Issue by gowcage** [protocolbuffers/protobuf#5639]
error info:
[libprotobuf FATAL google/protobuf/stubs/common.cc:78] This program was compiled against version 3.0.0 of the Protocol Buffer runtime library, which is not compatible with the installed version (3.3.0).  Contact the program author for an update.  If you compiled the program yourself, make sure that your headers are from the same version of Protocol Buffers as your link-time library.  (Version verification failed in "out/soong/.intermediates/frameworks/av/drm/libmediadrm/libmediadrm/android_arm_armv7-a-neon_cortex-a15_shared_core/gen/proto/frameworks/av/drm/libmediadrm/protos/plugin_metrics.pb.cc".)terminate called after throwing an instance of 'google::protobuf::FatalException'
  what():  This program was compiled against version 3.0.0 of the Protocol Buffer runtime library, which is not compatible with the installed version (3.3.0).  Contact the program author for an update.  If you compiled the program yourself, make sure that your headers are from the same version of Protocol Buffers as your link-time library.  (Version verification failed in "out/soong/.intermediates/frameworks/av/drm/libmediadrm/libmediadrm/android_arm_armv7-a-neon_cortex-a15_shared_core/gen/proto/frameworks/av/drm/libmediadrm/protos/plugin_metrics.pb.cc".)
Aborted

I am running on Android 8.1 after compiling, the system prompts this error.
Can anyone help me see what the problem is? Grateful！

**Comment by acozzette** [comment]
It looks like you built your protobuf generated code with `protoc` 3.0 but you're trying to link it against `libprotobuf` 3.3. These versions have to match, so probably the easiest fix would be to rebuild your generated protobuf code (i.e. the `pb.h` and `pb.cc` files) with `protoc` 3.3.

**Comment by suwenqiang** [comment]
> error info:
> [libprotobuf FATAL google/protobuf/stubs/common.cc:78] This program was compiled against version 3.0.0 of the Protocol Buffer runtime library, which is not compatible with the installed version (3.3.0). Contact the program author for an update. If you compiled the program yourself, make sure that your headers are from the same version of Protocol Buffers as your link-time library. (Version verification failed in "out/soong/.intermediates/frameworks/av/drm/libmediadrm/libmediadrm/android_arm_armv7-a-neon_cortex-a15_shared_core/gen/proto/frameworks/av/drm/libmediadrm/protos/plugin_metrics.pb.cc".)terminate called after throwing an instance of 'google::protobuf::FatalException'
> what(): This program was compiled against version 3.0.0 of the Protocol Buffer runtime library, which is not compatible with the installed version (3.3.0). Contact the program author for an update. If you compiled the program yourself, make sure that your headers are from the same version of Protocol Buffers as your link-time library. (Version verification failed in "out/soong/.intermediates/frameworks/av/drm/libmediadrm/libmediadrm/android_arm_armv7-a-neon_cortex-a15_shared_core/gen/proto/frameworks/av/drm/libmediadrm/protos/plugin_metrics.pb.cc".)
> Aborted
> 
> I am running on Android 8.1 after compiling, the system prompts this error.
> Can anyone help me see what the problem is? Grateful！

can you tell me how to fix this problem at last?

**Comment by murphypei** [comment]
> It looks like you built your protobuf generated code with `protoc` 3.0 but you're trying to link it against `libprotobuf` 3.3. These versions have to match, so probably the easiest fix would be to rebuild your generated protobuf code (i.e. the `pb.h` and `pb.cc` files) with `protoc` 3.3.

I think the problem is not that simple. I built my protobuf generate code with `protoc` 3.6.1 and link it against `libprotobuf` 3.6.1, this problem still appeared **on some android devices**. I debugged the program and found that there is a `/system/lib64/libprotobuf-full-cpp.so` library and it has GOOGLE_PROTOBUF_VERSION=3.0.0, I think this symbol is not covered by the android `libprotobuf-3.6.1`. By the way, I haven't found a solution yet.
